bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+local goipath="${1}"
+local file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
+             "${gopath}/src/${goipath}" \
+             "${prefix}${gopath}/src/${goipath}"; do
+  [[ ${file}/ == ${proot}/* ]] && file=".${file#${proot}}"
```

Comment:
Please more comments on what is the code supposed to do. My interpretation: if the file path is prefixed by any root import path, trim the import path from the left. Well written code in a lot of languages does not need comments. Though, this is bash and it is not so easy for non-bash oriented person to interpret it properly (with all its consequences).

What is the dot in the `file=".${file#${proot}}"` for?

Reply:
That is just a normalization routine to find the local file name regardless of how it was passed

The root of the change of course is that golist does not output relative file names and there are many ways to write absolute file names due to the symlinking go forces on us. While I'm it I normalized all path variants to avoid problems

They are transformed in the `./something` relative paths the `find . something -iname *.go` used to generate and that the original routine was written around. This is the most robust way to create something realpath will always transform in something we can handle.